test/helper/mockGetBoundingClientRect.ts

```diff
@@ -17,10 +18,17 @@ export function mockGetBoundingClientRect(rect: Partial<DOMRect>) {
     right: 0,
     bottom: 0,
     left: 0,
-    toJSON: () => '{}',
     ...rect,
+  };
+  Element.prototype.getBoundingClientRect = () => ({
+    ...mockDomRect,
+    toJSON: () => JSON.stringify(mockDomRect),
```

Comment:
Vitest will use this method to print difference if a test fails. So having the hardcoded `{}` was not helping.

Reply:
who knew, I'll put that one in my back pocket for later